tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenWithTransactionTest.cs

```diff
@@ -0,0 +1,150 @@
+                case "callback":
+                    {
+                        var valueDocument = value.AsBsonDocument;
+                        JsonDrivenHelper.EnsureAllFieldsAreValid(valueDocument, "operations");
+                        var operations = valueDocument["operations"].AsBsonArray;
+                        _callbackAsync = ParseCallbackAsync(operations);
+                        _callback = ParseCallback(operations);
+                    }
+                    return;
+            }
+
+            base.SetArgument(name, value);
+        }
+
+        // private methods
+        private TransactionOptions ParseOptions(BsonDocument document)
+        {
+            JsonDrivenHelper.EnsureAllFieldsAreValid(document, "readConcern", "readPreference", "writeConcern");
+
+            var options = new TransactionOptions();
+            if (document.Contains("readConcern"))
+            {
+                options = options.With(readConcern: ReadConcern.FromBsonDocument(document["readConcern"].AsBsonDocument));
+            }
+
+            if (document.Contains("readPreference"))
+            {
+                options = options.With(readPreference: ReadPreference.FromBsonDocument(document["readPreference"].AsBsonDocument));
+            }
+
+            if (document.Contains("writeConcern"))
+            {
+                options = options.With(writeConcern: WriteConcern.FromBsonDocument(document["writeConcern"].AsBsonDocument));
+            }
+
+            return options;
+        }
+
+        private Func<IClientSessionHandle, CancellationToken, BsonDocument> ParseCallback(BsonArray operations)
+        {
+            return (session, cancelationToken) =>
+            {
+                return ParseCallbackOperations(operations, false);
+            };
+        }
+
+        private Func<IClientSessionHandle, CancellationToken, Task<BsonDocument>> ParseCallbackAsync(BsonArray operations)
+        {
+            return (session, cancelationToken) =>
+            {
+                var result = ParseCallbackOperations(operations, true);
+                return Task.FromResult(result);
+            };
+        }
+
+        private BsonDocument ParseCallbackOperations(BsonArray operations, bool async)
+        {
+            foreach (var operation in operations.Cast<BsonDocument>())
+            {
+                var methodName = operation["name"].AsString;
+                var @object = operation["object"].AsString;
+                var test = _jsonDrivenTestFactory.CreateTest(@object, methodName);
+                test.Arrange(operation);
+                if (async)
+                {
+                    test.ActAsync(CancellationToken.None).GetAwaiter().GetResult();
+                }
+                else
+                {
+                    test.Act(CancellationToken.None);
+                }
+                test.ThrowActualExceptionIfNotNull();
+            }
+            return _result;
+        }
```

Comment:
Done.
Note: `var tests = new List<JsonDrivenTest>();` should be inside callback to have new internal fields values for each test. Otherwise, the results of previous steps will affect current